components/codec/src/byte.rs

```diff
@@ -341,12 +344,286 @@ impl MemComparableCodecHelper for DescendingMemComparableCodecHelper {
+            (
+                vec![0, 0, 0, 0, 0, 0, 0, 0],
+                vec![0, 0, 0, 0, 0, 0, 0, 0, 255, 0, 0, 0, 0, 0, 0, 0, 0, 247],
+            ),
+            (
+                vec![1, 2, 3, 4, 5, 6, 7, 8],
+                vec![1, 2, 3, 4, 5, 6, 7, 8, 255, 0, 0, 0, 0, 0, 0, 0, 0, 247],
+            ),
+            (
+                vec![1, 2, 3, 4, 5, 6, 7, 8, 9],
+                vec![1, 2, 3, 4, 5, 6, 7, 8, 255, 9, 0, 0, 0, 0, 0, 0, 0, 248],
+            ),
+        ];
+        for (src, encoded) in cases {
+            assert_eq!(encoded.as_slice().read_bytes().unwrap(), src);
+        }
+    }
+
+    #[test]
+    fn test_encode_compact_byte_len() {
+        let cases = vec![
+            (1, vec![0]),
+            (6, vec![10, 104, 101, 108, 108, 111]),
+            (6, vec![10, 104, 101, 108, 108, 111, 222]),
+            (7, vec![12, 228, 184, 150, 231, 149, 140]),
+            (7, vec![12, 228, 184, 150, 231, 149, 140, 222]),
+        ];
+        for (len, case) in cases {
+            assert_eq!(len, encoded_compact_len(&case).unwrap());
+        }
+    }
+
+    #[test]
+    fn test_compact_codec() {
+        use super::{CompactByteDecoder, CompactByteEncoder};
+        let tests = vec!["", "hello", "世界"];
+        for &s in &tests {
+            let max_size = s.len() + number::MAX_VARINT64_LENGTH;
+            let mut buf = Vec::with_capacity(max_size);
+            buf.write_compact_bytes(s.as_bytes()).unwrap();
+            assert!(buf.len() <= max_size);
+            let mut input = buf.as_slice();
+            let decoded = input.read_compact_bytes().unwrap();
+            assert!(input.is_empty());
+            assert_eq!(decoded, s.as_bytes());
+        }
+    }
+
+    #[test]
+    fn test_write_compact_bytes_for_file() {
+        use std::{env, fs, fs::File};
+        let cases = vec![
+            ("", vec![0]),
+            ("hello", vec![10, 104, 101, 108, 108, 111]),
+            ("世界", vec![12, 228, 184, 150, 231, 149, 140]),
+        ];
+        for (s, exp) in cases {
+            let mut path = env::temp_dir();
+            path.push("write-compact-codec-file");
+            {
+                let mut f = File::create(&path).unwrap();
+                f.write_compact_bytes(s.as_bytes()).unwrap();
+            }
+            assert_eq!(fs::read(&path).unwrap(), exp);
+            fs::remove_file(path).unwrap();
+        }
+    }
+
+    #[test]
+    fn test_read_compact_bytes_for_file() {
+        use std::{env, fs, fs::File, io::BufReader};
+        let cases = vec![
+            ("", vec![0]),
+            ("hello", vec![10, 104, 101, 108, 108, 111]),
+            ("世界", vec![12, 228, 184, 150, 231, 149, 140]),
+        ];
+        for (exp, encoded) in cases {
+            let mut path = env::temp_dir();
+            path.push("read-compact-codec-file");
+            fs::write(&path, &encoded).unwrap();
+            let f = File::open(&path).unwrap();
```

Comment:
Could we add a test to write all data in the `cases` into one file?

Reply:
hi, did you mean to add a  test case for file contains multiple compact bytes, and decode the first one? 
I add one for this case,  pease take a look, thanks